File: agent/src/main/java/com/smancode/smanagent/tools/BatchSubResult.java
package com.smancode.smanagent.tools;

/**
 * 批量工具的子结果
 * <p>
 * 用于 batch 工具记录每个子工具的执行结果
 */
public class BatchSubResult {
    /**
     * 子工具名称
     */
    private String toolName;

    /**
     * 是否成功
     */
    private boolean success;

    /**
     * 工具结果（复用 ToolResult）
     */
    private ToolResult result;

    /**
     * 错误信息（如果失败）
     */
    private String error;

    public BatchSubResult() {
    }

    public BatchSubResult(String toolName, boolean success, ToolResult result, String error) {
        this.toolName = toolName;
        this.success = success;
        this.result = result;
        this.error = error;
    }

    public String getToolName() {
        return toolName;
    }

    public void setToolName(String toolName) {
        this.toolName = toolName;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public ToolResult getResult() {
        return result;
    }

    public void setResult(ToolResult result) {
        this.result = result;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }
}

File: src/main/kotlin/com/smancode/sman/tools/BatchSubResult.kt
package com.smancode.sman.tools

/**
 * 批量工具的子结果
 *
 * 用于 batch 工具记录每个子工具的执行结果
 */
class BatchSubResult {

    /**
     * 子工具名称
     */
    var toolName: String? = null

    /**
     * 是否成功
     */
    var isSuccess: Boolean = false

    /**
     * 工具结果（复用 ToolResult）
     */
    var result: ToolResult? = null

    /**
     * 错误信息（如果失败）
     */
    var error: String? = null

    constructor()

    constructor(toolName: String, success: Boolean, result: ToolResult?, error: String?) {
        this.toolName = toolName
        this.isSuccess = success
        this.result = result
        this.error = error
    }
}
